File: jobs/java/spark-job/src/main/java/com/airscholar/spark/wordcount.java
package com.airscholar.spark;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import java.util.Arrays;
import java.util.List;
/**
 * Hello world!
 *
 */
public class wordcount 
{
    public static void main( String[] args )
    {
        SparkConf conf = new SparkConf().setAppName("WordCount").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);

        String text = "Hallo World! Hello Spark!";
        List<String> lines = Arrays.asList(text.split(" "));

        JavaRdd<String> rdd = sc.parallelize(lines);
        JavaPairRDD<String, Integer> wordCounts = rdd.mapToPair(word -> new Tuple2<>(word, 1))
                .reduceByKey((a, b) -> a + b);
        
        List<Tuple2<String, Integer>> output = wordCounts.collect();
        for (Tuple2<String, Integer> tuple : output) {
            System.out.println(tuple._1() + ": " + tuple._2());
        }
    }
}

File: jobs/scala/wordcount.scala
import org.apache.spark.sql.SparkSession

object WordCount {
    def main(args: Array[String]) {
        val spark = SparkSession.builder
            .appName("Word Count")
            .getOrCreate()
        val sc = spark.sparkContext

        val textData = sc.parallelize(List("Hello world.", "Hello Spark.", "Hello Airflow."))

        val counts = textData.flatMap(line => line.split(" "))
            .map(word => (word, 1))
            .reduceByKey(_ + _)

        counts.collect().foreach(println)

        spark.stop()
        
    }
}